alsaaudio.c

```diff
@@ -1106,16 +1107,20 @@ alsapcm_read(alsapcm_t *self, PyObject *args)
 	}
 
 	if (res <= 0) {
+		sizeout = 0;
+	} else {
+		sizeout = res * self->framesize;
+	}
+	
 #if PY_MAJOR_VERSION < 3
 		/* If the following fails, it will free the object */
```

Comment:
I'd prefer if the `Resize` was only done if `size != sizeout`. I'd expect Python to do the right thing and not actually do work if it isn't required, but I'd be defensive nonetheless

Reply:
OK, I'll make the suggested changes.